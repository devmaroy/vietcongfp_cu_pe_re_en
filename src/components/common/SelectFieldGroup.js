import React from 'react';
import PropTypes from 'prop-types';

const SelectFieldGroup = ( { name, options, className, onChange, value, info } ) => {
    return (
        <div>
            <select
                name={ name }
                value={ value }
                className={ className }
                onChange={ onChange }
            >
                { 
                    options.length > 0 && (
                        options.map( ( option, index ) => <option key={ index } value={ option.key }>{ option.text }</option> )
                    )
                }
            </select>
            { info && <small>{ info }</small> }
        </div>
    );
};

SelectFieldGroup.propTypes = {
    name: PropTypes.string.isRequired,
    options: PropTypes.array.isRequired,
    value: PropTypes.string.isRequired,
    info: PropTypes.string,
    onChange: PropTypes.func.isRequired,
    className: PropTypes.string
};

export default SelectFieldGroup;