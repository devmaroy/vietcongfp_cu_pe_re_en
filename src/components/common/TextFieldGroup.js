import React from 'react';
import PropTypes from 'prop-types';

const TextFieldGroup = ( { name, placeholder, value, error, info, type, onChange, disabled, className } ) => {
    return (
        <div>
            <input
                type={ type }
                placeholder={ placeholder }
                name={ name }
                value={ value }
                onChange={ onChange }
                disabled={ disabled }
                className={ className }
            />
            { info && <small>{ info }</small> }
            { error && <div>{ error }</div> }
        </div>
    );
};

TextFieldGroup.propTypes = {
    name: PropTypes.string.isRequired,
    placeholder: PropTypes.string,
    value: PropTypes.string.isRequired,
    info: PropTypes.string,
    error: PropTypes.string,
    type: PropTypes.string.isRequired,
    onChange: PropTypes.func.isRequired,
    disabled: PropTypes.bool,
    className: PropTypes.string
};
  
TextFieldGroup.defaultProps = {
    type: 'text'
};

export default TextFieldGroup;