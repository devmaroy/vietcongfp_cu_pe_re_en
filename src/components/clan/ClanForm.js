import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import TextFieldGroup from '../common/TextFieldGroup';
import TextAreaFieldGroup from '../common/TextAreaFieldGroup';
import SelectFieldGroup from '../common/SelectFieldGroup';
import validator from 'validator';
import Spinner from '../common/Spinner';
import moment from 'moment';
import countries from '../../utils/countries';

export class ClanForm extends React.Component {
    state = {
        tag: this.props.clan ? this.props.clan.tag : '',
        name: this.props.clan ? this.props.clan.name : '',
        country: this.props.clan ? this.props.clan.country : '',
        leader: this.props.clan ? this.props.clan.leader : '',
        url: this.props.clan ? this.props.clan.url : '',
        eslName: this.props.clan ? this.props.clan.eslName : '',
        eslUrl: this.props.clan ? this.props.clan.eslUrl : '',
        createdAt: this.props.clan ? this.props.clan.createdAt : moment(),
        errors: {}
    };

    handleValidation = () => {
        const errors = {};
        let formIsValid = true;

        // Tag
        if ( ! this.state.tag ) {
            formIsValid = false;
            errors[ 'tag' ] = 'Tag cannot be empty';
        }

        // Name
        if ( ! this.state.name ) {
            formIsValid = false;
            errors[ 'name' ] = 'Name cannot be empty';
        }

        // Country
        if ( ! this.state.country ) {
            formIsValid = false;
            errors[ 'country' ] = 'Country must be provided';
        }

        // Leader
        if ( ! this.state.leader ) {
            formIsValid = false;
            errors[ 'leader' ] = 'Leader cannot be empty';
        }

        // Url
        if ( ! this.state.url ) {
            formIsValid = false;
            errors[ 'url' ] = 'Url cannot be empty';
        } else if ( ! validator.isURL( this.state.url )  ) {
            formIsValid = false;
            errors[ 'url' ] = 'Url must be valid';
        }

        // ESL Url
        if ( this.state.eslUrl ) {
            if ( ! validator.isURL( this.state.eslUrl ) ) {
                formIsValid = false;
                errors[ 'eslUrl' ] = 'ESL url must be valid';
            }
        }

        this.setState( () => ( { errors } ) );

        return formIsValid;
    };

    handleChange = ( e ) => {
        const target = e.target;
        const value = target.value;
        const name = target.name;

        this.setState( () => ( { [ name ]: value } ) );
    };

    onSubmit = ( e ) => {
        e.preventDefault();

        if ( this.handleValidation() ) {
            const clanData = {
                tag: this.state.tag,
                name: this.state.name,
                country: this.state.country,
                leader: this.state.leader,
                url: this.state.url,
                eslName: this.state.eslName,
                eslUrl: this.state.eslUrl,
                createdAt: this.state.createdAt.valueOf()
            };
            
            this.props.onSubmit( clanData );
        }
    }

    render() {
        const { errors } = this.state;

        return (
            <form onSubmit={ this.onSubmit } className="form" noValidate>
                <div className="input-group">
                    <div className="input-group__item">
                        <TextFieldGroup 
                            name="tag"
                            placeholder="Tag"
                            value={ this.state.tag }
                            onChange={ this.handleChange }
                            error={ errors.tag }
                            className="text-input"
                        />
                    </div>
                </div>
                <div className="input-group">
                    <div className="input-group__item">
                        <TextFieldGroup 
                            name="name"
                            placeholder="Name"
                            value={ this.state.name }
                            onChange={ this.handleChange }
                            error={ errors.name }
                            className="text-input"
                        />
                    </div>
                </div>
                <div className="input-group">
                    <div className="input-group__item">
                        <SelectFieldGroup
                            name="country"
                            value={ this.state.country }
                            options={ countries }
                            onChange={ this.handleChange }
                            className="select"
                        />
                    </div>
                </div>
                <div className="input-group">
                    <div className="input-group__item">
                        <TextFieldGroup 
                            name="leader"
                            placeholder="Leader"
                            value={ this.state.leader }
                            onChange={ this.handleChange }
                            error={ errors.leader }
                            className="text-input"
                        />
                    </div>
                </div>
                <div className="input-group">
                    <div className="input-group__item">
                        <TextFieldGroup 
                            name="url"
                            type="url"
                            placeholder="Website"
                            value={ this.state.url }
                            onChange={ this.handleChange }
                            error={ errors.url }
                            className="text-input"
                        />
                    </div>
                </div>
                <div className="input-group">
                    <div className="input-group__item">
                        <TextFieldGroup 
                            name="eslName"
                            placeholder="ESL Clan name"
                            value={ this.state.eslName }
                            onChange={ this.handleChange }
                            error={ errors.eslName }
                            className="text-input"
                        />
                    </div>
                </div>
                <div className="input-group">
                    <div className="input-group__item">
                        <TextFieldGroup 
                            name="eslUrl"
                            type="url"
                            placeholder="ESL Clan website"
                            value={ this.state.eslUrl }
                            onChange={ this.handleChange }
                            error={ errors.eslUrl }
                            className="text-input"
                        />
                    </div>
                </div>
                <div className="form__meta">
                    <button className="button">Save Clan</button>
                </div>
            </form>
        );
    };
};

ClanForm.propTypes = {
    clan: PropTypes.object,
    onSubmit: PropTypes.func.isRequired
};

export default ClanForm;