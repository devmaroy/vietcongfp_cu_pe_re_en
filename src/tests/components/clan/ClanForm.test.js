import React from 'react';
import { shallow } from 'enzyme';
import ClanForm from '../../../components/clan/ClanForm';
import TextFieldGroup from '../../../components/common/TextFieldGroup';
import SelectFieldGroup from '../../../components/common/SelectFieldGroup';
import clans from '../../fixtures/clans';

test( 'should correctly render form inside ClanForm', () => {
    const wrapper = shallow( 
        <ClanForm 
            onSubmit={ () => {} }
        />
    );

    expect( wrapper ).toMatchSnapshot();
    expect( wrapper.find( 'form' ).exists() ).toBeTruthy();
});

test( 'should correctly render error object if no data provided', () => {
    const wrapper = shallow(
        <ClanForm
            onSubmit={ () => {} }
        />
    );

    wrapper.find( 'form' ).simulate( 'submit', {
        preventDefault: () => {}
    });

    expect( wrapper.state( 'errors' ).tag.length ).toBeGreaterThan( 0 );
    expect( wrapper.state( 'errors' ).name.length ).toBeGreaterThan( 0 );
    expect( wrapper.state( 'errors' ).country.length ).toBeGreaterThan( 0 );
    expect( wrapper.state( 'errors' ).leader.length ).toBeGreaterThan( 0 );
    expect( wrapper.state( 'errors' ).url.length ).toBeGreaterThan( 0 );
});

test( 'should set tag on change', () => {
    const value = 'AVP';

    const wrapper = shallow( 
        <ClanForm
            onSubmit={ () => {} }
        />
    );

    const event = { target: { name: 'tag', value } };

    wrapper.find( TextFieldGroup ).at( 0 ).simulate( 'change', event );

    expect( wrapper.state( 'tag' ) ).toBe( value );
});

test( 'should set name on change', () => {
    const value = 'Awesome Vietcong Players';

    const wrapper = shallow( 
        <ClanForm
            onSubmit={ () => {} }
        />
    );

    const event = { target: { name: 'name', value } };

    wrapper.find( TextFieldGroup ).at( 1 ).simulate( 'change', event );

    expect( wrapper.state( 'name' ) ).toBe( value );
});

test( 'should set tag on change', () => {
    const value = 'CZ';

    const wrapper = shallow( 
        <ClanForm
            onSubmit={ () => {} }
        />
    );

    const event = { target: { name: 'country', value } };

    wrapper.find( SelectFieldGroup ).simulate( 'change', event );

    expect( wrapper.state( 'country' ) ).toBe( value );
});

test( 'should set leader on change', () => {
    const value = 'Filip9';

    const wrapper = shallow( 
        <ClanForm
            onSubmit={ () => {} }
        />
    );

    const event = { target: { name: 'leader', value } };

    wrapper.find( TextFieldGroup ).at( 2 ).simulate( 'change', event );

    expect( wrapper.state( 'leader' ) ).toBe( value );
});

test( 'should set url on change', () => {
    const value = 'https://avp.wz.cz/';

    const wrapper = shallow( 
        <ClanForm
            onSubmit={ () => {} }
        />
    );

    const event = { target: { name: 'url', value } };

    wrapper.find( TextFieldGroup ).at( 3 ).simulate( 'change', event );

    expect( wrapper.state( 'url' ) ).toBe( value );
});

test( 'should set eslName on change', () => {
    const value = 'AVP-CZ-SK';

    const wrapper = shallow( 
        <ClanForm
            onSubmit={ () => {} }
        />
    );

    const event = { target: { name: 'eslName', value } };

    wrapper.find( TextFieldGroup ).at( 4 ).simulate( 'change', event );

    expect( wrapper.state( 'eslName' ) ).toBe( value );
});

test( 'should set eslUrl on change', () => {
    const value = 'https://esl.com/team/96';

    const wrapper = shallow( 
        <ClanForm
            onSubmit={ () => {} }
        />
    );

    const event = { target: { name: 'eslUrl', value } };

    wrapper.find( TextFieldGroup ).at( 5 ).simulate( 'change', event );

    expect( wrapper.state( 'eslUrl' ) ).toBe( value );
});

test( 'should call onSubmit prop for valid form submission', () => {
    const onSubmitSpy = jest.fn();

    const wrapper = shallow(
        <ClanForm 
            clan={ clans[0] }
            onSubmit={ onSubmitSpy }
        />
    );

    wrapper.find( 'form' ).simulate( 'submit', {
        preventDefault: () => {}
    });

    expect( wrapper.state( 'errors' ) ).toEqual( {} );

    expect( onSubmitSpy ).toHaveBeenLastCalledWith({
        tag: clans[0].tag,
        name: clans[0].name,
        country: clans[0].country,
        leader: clans[0].leader,
        url: clans[0].url,
        eslName: clans[0].eslName,
        eslUrl: clans[0].eslUrl,
        createdAt: clans[0].createdAt
    });
});