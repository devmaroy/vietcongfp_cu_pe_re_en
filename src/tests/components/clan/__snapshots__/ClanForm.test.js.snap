// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render form inside ClanForm 1`] = `
<form
  className="form"
  noValidate={true}
  onSubmit={[Function]}
>
  <div
    className="input-group"
  >
    <div
      className="input-group__item"
    >
      <TextFieldGroup
        className="text-input"
        name="tag"
        onChange={[Function]}
        placeholder="Tag"
        type="text"
        value=""
      />
    </div>
  </div>
  <div
    className="input-group"
  >
    <div
      className="input-group__item"
    >
      <TextFieldGroup
        className="text-input"
        name="name"
        onChange={[Function]}
        placeholder="Name"
        type="text"
        value=""
      />
    </div>
  </div>
  <div
    className="input-group"
  >
    <div
      className="input-group__item"
    >
      <SelectFieldGroup
        className="select"
        name="country"
        onChange={[Function]}
        options={
          Array [
            Object {
              "key": "AF",
              "text": "Afghanistan",
            },
            Object {
              "key": "AX",
              "text": "Ã…land Islands",
            },
            Object {
              "key": "AL",
              "text": "Albania",
            },
            Object {
              "key": "DZ",
              "text": "Algeria",
            },
            Object {
              "key": "AS",
              "text": "American Samoa",
            },
            Object {
              "key": "AD",
              "text": "Andorra",
            },
            Object {
              "key": "AO",
              "text": "Angola",
            },
            Object {
              "key": "AI",
              "text": "Anguilla",
            },
            Object {
              "key": "AQ",
              "text": "Antarctica",
            },
            Object {
              "key": "AG",
              "text": "Antigua and Barbuda",
            },
            Object {
              "key": "AR",
              "text": "Argentina",
            },
            Object {
              "key": "AM",
              "text": "Armenia",
            },
            Object {
              "key": "AW",
              "text": "Aruba",
            },
            Object {
              "key": "AU",
              "text": "Australia",
            },
            Object {
              "key": "AT",
              "text": "Austria",
            },
            Object {
              "key": "AZ",
              "text": "Azerbaijan",
            },
            Object {
              "key": "BS",
              "text": "Bahamas",
            },
            Object {
              "key": "BH",
              "text": "Bahrain",
            },
            Object {
              "key": "BD",
              "text": "Bangladesh",
            },
            Object {
              "key": "BB",
              "text": "Barbados",
            },
            Object {
              "key": "BY",
              "text": "Belarus",
            },
            Object {
              "key": "BE",
              "text": "Belgium",
            },
            Object {
              "key": "BZ",
              "text": "Belize",
            },
            Object {
              "key": "BJ",
              "text": "Benin",
            },
            Object {
              "key": "BM",
              "text": "Bermuda",
            },
            Object {
              "key": "BT",
              "text": "Bhutan",
            },
            Object {
              "key": "BO",
              "text": "Bolivia",
            },
            Object {
              "key": "BA",
              "text": "Bosnia and Herzegovina",
            },
            Object {
              "key": "BW",
              "text": "Botswana",
            },
            Object {
              "key": "BV",
              "text": "Bouvet Island",
            },
            Object {
              "key": "BR",
              "text": "Brazil",
            },
            Object {
              "key": "IO",
              "text": "British Indian Ocean Territory",
            },
            Object {
              "key": "BN",
              "text": "Brunei Darussalam",
            },
            Object {
              "key": "BG",
              "text": "Bulgaria",
            },
            Object {
              "key": "BF",
              "text": "Burkina Faso",
            },
            Object {
              "key": "BI",
              "text": "Burundi",
            },
            Object {
              "key": "KH",
              "text": "Cambodia",
            },
            Object {
              "key": "CM",
              "text": "Cameroon",
            },
            Object {
              "key": "CA",
              "text": "Canada",
            },
            Object {
              "key": "CV",
              "text": "Cape Verde",
            },
            Object {
              "key": "KY",
              "text": "Cayman Islands",
            },
            Object {
              "key": "CF",
              "text": "Central African Republic",
            },
            Object {
              "key": "TD",
              "text": "Chad",
            },
            Object {
              "key": "CL",
              "text": "Chile",
            },
            Object {
              "key": "CN",
              "text": "China",
            },
            Object {
              "key": "CX",
              "text": "Christmas Island",
            },
            Object {
              "key": "CC",
              "text": "Cocos (Keeling) Islands",
            },
            Object {
              "key": "CO",
              "text": "Colombia",
            },
            Object {
              "key": "KM",
              "text": "Comoros",
            },
            Object {
              "key": "CG",
              "text": "Congo",
            },
            Object {
              "key": "CD",
              "text": "Congo, The Democratic Republic of the",
            },
            Object {
              "key": "CK",
              "text": "Cook Islands",
            },
            Object {
              "key": "CR",
              "text": "Costa Rica",
            },
            Object {
              "key": "CI",
              "text": "Cote D'Ivoire",
            },
            Object {
              "key": "HR",
              "text": "Croatia",
            },
            Object {
              "key": "CU",
              "text": "Cuba",
            },
            Object {
              "key": "CY",
              "text": "Cyprus",
            },
            Object {
              "key": "CZ",
              "text": "Czech Republic",
            },
            Object {
              "key": "DK",
              "text": "Denmark",
            },
            Object {
              "key": "DJ",
              "text": "Djibouti",
            },
            Object {
              "key": "DM",
              "text": "Dominica",
            },
            Object {
              "key": "DO",
              "text": "Dominican Republic",
            },
            Object {
              "key": "EC",
              "text": "Ecuador",
            },
            Object {
              "key": "EG",
              "text": "Egypt",
            },
            Object {
              "key": "SV",
              "text": "El Salvador",
            },
            Object {
              "key": "GQ",
              "text": "Equatorial Guinea",
            },
            Object {
              "key": "ER",
              "text": "Eritrea",
            },
            Object {
              "key": "EE",
              "text": "Estonia",
            },
            Object {
              "key": "ET",
              "text": "Ethiopia",
            },
            Object {
              "key": "FK",
              "text": "Falkland Islands (Malvinas)",
            },
            Object {
              "key": "FO",
              "text": "Faroe Islands",
            },
            Object {
              "key": "FJ",
              "text": "Fiji",
            },
            Object {
              "key": "FI",
              "text": "Finland",
            },
            Object {
              "key": "FR",
              "text": "France",
            },
            Object {
              "key": "GF",
              "text": "French Guiana",
            },
            Object {
              "key": "PF",
              "text": "French Polynesia",
            },
            Object {
              "key": "TF",
              "text": "French Southern Territories",
            },
            Object {
              "key": "GA",
              "text": "Gabon",
            },
            Object {
              "key": "GM",
              "text": "Gambia",
            },
            Object {
              "key": "GE",
              "text": "Georgia",
            },
            Object {
              "key": "DE",
              "text": "Germany",
            },
            Object {
              "key": "GH",
              "text": "Ghana",
            },
            Object {
              "key": "GI",
              "text": "Gibraltar",
            },
            Object {
              "key": "GR",
              "text": "Greece",
            },
            Object {
              "key": "GL",
              "text": "Greenland",
            },
            Object {
              "key": "GD",
              "text": "Grenada",
            },
            Object {
              "key": "GP",
              "text": "Guadeloupe",
            },
            Object {
              "key": "GU",
              "text": "Guam",
            },
            Object {
              "key": "GT",
              "text": "Guatemala",
            },
            Object {
              "key": "GG",
              "text": "Guernsey",
            },
            Object {
              "key": "GN",
              "text": "Guinea",
            },
            Object {
              "key": "GW",
              "text": "Guinea-Bissau",
            },
            Object {
              "key": "GY",
              "text": "Guyana",
            },
            Object {
              "key": "HT",
              "text": "Haiti",
            },
            Object {
              "key": "HM",
              "text": "Heard Island and Mcdonald Islands",
            },
            Object {
              "key": "VA",
              "text": "Holy See (Vatican City State)",
            },
            Object {
              "key": "HN",
              "text": "Honduras",
            },
            Object {
              "key": "HK",
              "text": "Hong Kong",
            },
            Object {
              "key": "HU",
              "text": "Hungary",
            },
            Object {
              "key": "IS",
              "text": "Iceland",
            },
            Object {
              "key": "IN",
              "text": "India",
            },
            Object {
              "key": "ID",
              "text": "Indonesia",
            },
            Object {
              "key": "IR",
              "text": "Iran, Islamic Republic Of",
            },
            Object {
              "key": "IQ",
              "text": "Iraq",
            },
            Object {
              "key": "IE",
              "text": "Ireland",
            },
            Object {
              "key": "IM",
              "text": "Isle of Man",
            },
            Object {
              "key": "IL",
              "text": "Israel",
            },
            Object {
              "key": "IT",
              "text": "Italy",
            },
            Object {
              "key": "JM",
              "text": "Jamaica",
            },
            Object {
              "key": "JP",
              "text": "Japan",
            },
            Object {
              "key": "JE",
              "text": "Jersey",
            },
            Object {
              "key": "JO",
              "text": "Jordan",
            },
            Object {
              "key": "KZ",
              "text": "Kazakhstan",
            },
            Object {
              "key": "KE",
              "text": "Kenya",
            },
            Object {
              "key": "KI",
              "text": "Kiribati",
            },
            Object {
              "key": "KP",
              "text": "Korea, Democratic People'S Republic of",
            },
            Object {
              "key": "KR",
              "text": "Korea, Republic of",
            },
            Object {
              "key": "KW",
              "text": "Kuwait",
            },
            Object {
              "key": "KG",
              "text": "Kyrgyzstan",
            },
            Object {
              "key": "LA",
              "text": "Lao People'S Democratic Republic",
            },
            Object {
              "key": "LV",
              "text": "Latvia",
            },
            Object {
              "key": "LB",
              "text": "Lebanon",
            },
            Object {
              "key": "LS",
              "text": "Lesotho",
            },
            Object {
              "key": "LR",
              "text": "Liberia",
            },
            Object {
              "key": "LY",
              "text": "Libyan Arab Jamahiriya",
            },
            Object {
              "key": "LI",
              "text": "Liechtenstein",
            },
            Object {
              "key": "LT",
              "text": "Lithuania",
            },
            Object {
              "key": "LU",
              "text": "Luxembourg",
            },
            Object {
              "key": "MO",
              "text": "Macao",
            },
            Object {
              "key": "MK",
              "text": "Macedonia, The Former Yugoslav Republic of",
            },
            Object {
              "key": "MG",
              "text": "Madagascar",
            },
            Object {
              "key": "MW",
              "text": "Malawi",
            },
            Object {
              "key": "MY",
              "text": "Malaysia",
            },
            Object {
              "key": "MV",
              "text": "Maldives",
            },
            Object {
              "key": "ML",
              "text": "Mali",
            },
            Object {
              "key": "MT",
              "text": "Malta",
            },
            Object {
              "key": "MH",
              "text": "Marshall Islands",
            },
            Object {
              "key": "MQ",
              "text": "Martinique",
            },
            Object {
              "key": "MR",
              "text": "Mauritania",
            },
            Object {
              "key": "MU",
              "text": "Mauritius",
            },
            Object {
              "key": "YT",
              "text": "Mayotte",
            },
            Object {
              "key": "MX",
              "text": "Mexico",
            },
            Object {
              "key": "FM",
              "text": "Micronesia, Federated States of",
            },
            Object {
              "key": "MD",
              "text": "Moldova, Republic of",
            },
            Object {
              "key": "MC",
              "text": "Monaco",
            },
            Object {
              "key": "MN",
              "text": "Mongolia",
            },
            Object {
              "key": "MS",
              "text": "Montserrat",
            },
            Object {
              "key": "MA",
              "text": "Morocco",
            },
            Object {
              "key": "MZ",
              "text": "Mozambique",
            },
            Object {
              "key": "MM",
              "text": "Myanmar",
            },
            Object {
              "key": "NA",
              "text": "Namibia",
            },
            Object {
              "key": "NR",
              "text": "Nauru",
            },
            Object {
              "key": "NP",
              "text": "Nepal",
            },
            Object {
              "key": "NL",
              "text": "Netherlands",
            },
            Object {
              "key": "AN",
              "text": "Netherlands Antilles",
            },
            Object {
              "key": "NC",
              "text": "New Caledonia",
            },
            Object {
              "key": "NZ",
              "text": "New Zealand",
            },
            Object {
              "key": "NI",
              "text": "Nicaragua",
            },
            Object {
              "key": "NE",
              "text": "Niger",
            },
            Object {
              "key": "NG",
              "text": "Nigeria",
            },
            Object {
              "key": "NU",
              "text": "Niue",
            },
            Object {
              "key": "NF",
              "text": "Norfolk Island",
            },
            Object {
              "key": "MP",
              "text": "Northern Mariana Islands",
            },
            Object {
              "key": "NO",
              "text": "Norway",
            },
            Object {
              "key": "OM",
              "text": "Oman",
            },
            Object {
              "key": "PK",
              "text": "Pakistan",
            },
            Object {
              "key": "PW",
              "text": "Palau",
            },
            Object {
              "key": "PS",
              "text": "Palestinian Territory, Occupied",
            },
            Object {
              "key": "PA",
              "text": "Panama",
            },
            Object {
              "key": "PG",
              "text": "Papua New Guinea",
            },
            Object {
              "key": "PY",
              "text": "Paraguay",
            },
            Object {
              "key": "PE",
              "text": "Peru",
            },
            Object {
              "key": "PH",
              "text": "Philippines",
            },
            Object {
              "key": "PN",
              "text": "Pitcairn",
            },
            Object {
              "key": "PL",
              "text": "Poland",
            },
            Object {
              "key": "PT",
              "text": "Portugal",
            },
            Object {
              "key": "PR",
              "text": "Puerto Rico",
            },
            Object {
              "key": "QA",
              "text": "Qatar",
            },
            Object {
              "key": "RE",
              "text": "Reunion",
            },
            Object {
              "key": "RO",
              "text": "Romania",
            },
            Object {
              "key": "RU",
              "text": "Russian Federation",
            },
            Object {
              "key": "RW",
              "text": "RWANDA",
            },
            Object {
              "key": "SH",
              "text": "Saint Helena",
            },
            Object {
              "key": "KN",
              "text": "Saint Kitts and Nevis",
            },
            Object {
              "key": "LC",
              "text": "Saint Lucia",
            },
            Object {
              "key": "PM",
              "text": "Saint Pierre and Miquelon",
            },
            Object {
              "key": "VC",
              "text": "Saint Vincent and the Grenadines",
            },
            Object {
              "key": "WS",
              "text": "Samoa",
            },
            Object {
              "key": "SM",
              "text": "San Marino",
            },
            Object {
              "key": "ST",
              "text": "Sao Tome and Principe",
            },
            Object {
              "key": "SA",
              "text": "Saudi Arabia",
            },
            Object {
              "key": "SN",
              "text": "Senegal",
            },
            Object {
              "key": "CS",
              "text": "Serbia and Montenegro",
            },
            Object {
              "key": "SC",
              "text": "Seychelles",
            },
            Object {
              "key": "SL",
              "text": "Sierra Leone",
            },
            Object {
              "key": "SG",
              "text": "Singapore",
            },
            Object {
              "key": "SK",
              "text": "Slovakia",
            },
            Object {
              "key": "SI",
              "text": "Slovenia",
            },
            Object {
              "key": "SB",
              "text": "Solomon Islands",
            },
            Object {
              "key": "SO",
              "text": "Somalia",
            },
            Object {
              "key": "ZA",
              "text": "South Africa",
            },
            Object {
              "key": "GS",
              "text": "South Georgia and the South Sandwich Islands",
            },
            Object {
              "key": "ES",
              "text": "Spain",
            },
            Object {
              "key": "LK",
              "text": "Sri Lanka",
            },
            Object {
              "key": "SD",
              "text": "Sudan",
            },
            Object {
              "key": "SR",
              "text": "Suritext",
            },
            Object {
              "key": "SJ",
              "text": "Svalbard and Jan Mayen",
            },
            Object {
              "key": "SZ",
              "text": "Swaziland",
            },
            Object {
              "key": "SE",
              "text": "Sweden",
            },
            Object {
              "key": "CH",
              "text": "Switzerland",
            },
            Object {
              "key": "SY",
              "text": "Syrian Arab Republic",
            },
            Object {
              "key": "TW",
              "text": "Taiwan, Province of China",
            },
            Object {
              "key": "TJ",
              "text": "Tajikistan",
            },
            Object {
              "key": "TZ",
              "text": "Tanzania, United Republic of",
            },
            Object {
              "key": "TH",
              "text": "Thailand",
            },
            Object {
              "key": "TL",
              "text": "Timor-Leste",
            },
            Object {
              "key": "TG",
              "text": "Togo",
            },
            Object {
              "key": "TK",
              "text": "Tokelau",
            },
            Object {
              "key": "TO",
              "text": "Tonga",
            },
            Object {
              "key": "TT",
              "text": "Trinidad and Tobago",
            },
            Object {
              "key": "TN",
              "text": "Tunisia",
            },
            Object {
              "key": "TR",
              "text": "Turkey",
            },
            Object {
              "key": "TM",
              "text": "Turkmenistan",
            },
            Object {
              "key": "TC",
              "text": "Turks and Caicos Islands",
            },
            Object {
              "key": "TV",
              "text": "Tuvalu",
            },
            Object {
              "key": "UG",
              "text": "Uganda",
            },
            Object {
              "key": "UA",
              "text": "Ukraine",
            },
            Object {
              "key": "AE",
              "text": "United Arab Emirates",
            },
            Object {
              "key": "GB",
              "text": "United Kingdom",
            },
            Object {
              "key": "US",
              "text": "United States",
            },
            Object {
              "key": "UM",
              "text": "United States Minor Outlying Islands",
            },
            Object {
              "key": "UY",
              "text": "Uruguay",
            },
            Object {
              "key": "UZ",
              "text": "Uzbekistan",
            },
            Object {
              "key": "VU",
              "text": "Vanuatu",
            },
            Object {
              "key": "VE",
              "text": "Venezuela",
            },
            Object {
              "key": "VN",
              "text": "Viet Nam",
            },
            Object {
              "key": "VG",
              "text": "Virgin Islands, British",
            },
            Object {
              "key": "VI",
              "text": "Virgin Islands, U.S.",
            },
            Object {
              "key": "WF",
              "text": "Wallis and Futuna",
            },
            Object {
              "key": "EH",
              "text": "Western Sahara",
            },
            Object {
              "key": "YE",
              "text": "Yemen",
            },
            Object {
              "key": "ZM",
              "text": "Zambia",
            },
            Object {
              "key": "ZW",
              "text": "Zimbabwe",
            },
          ]
        }
        value=""
      />
    </div>
  </div>
  <div
    className="input-group"
  >
    <div
      className="input-group__item"
    >
      <TextFieldGroup
        className="text-input"
        name="leader"
        onChange={[Function]}
        placeholder="Leader"
        type="text"
        value=""
      />
    </div>
  </div>
  <div
    className="input-group"
  >
    <div
      className="input-group__item"
    >
      <TextFieldGroup
        className="text-input"
        name="url"
        onChange={[Function]}
        placeholder="Website"
        type="url"
        value=""
      />
    </div>
  </div>
  <div
    className="input-group"
  >
    <div
      className="input-group__item"
    >
      <TextFieldGroup
        className="text-input"
        name="eslName"
        onChange={[Function]}
        placeholder="ESL Clan name"
        type="text"
        value=""
      />
    </div>
  </div>
  <div
    className="input-group"
  >
    <div
      className="input-group__item"
    >
      <TextFieldGroup
        className="text-input"
        name="eslUrl"
        onChange={[Function]}
        placeholder="ESL Clan website"
        type="url"
        value=""
      />
    </div>
  </div>
  <div
    className="form__meta"
  >
    <button
      className="button"
    >
      Save Clan
    </button>
  </div>
</form>
`;
